pkg/render/packetcapture_api_test.go

```diff
@@ -0,0 +1,373 @@
+
+	// Rendering packet capture resources
+	var renderPacketCapture = func(i operator.InstallationSpec, config authentication.KeyValidatorConfig) (resources []client.Object) {
+		var pc = render.PacketCaptureAPI(
+			pullSecrets,
+			false,
+			&i,
+			config,
+			secret,
+			"",
+		)
+		Expect(pc.ResolveImages(nil)).To(BeNil())
+		resources, _ = pc.Objects()
+		return resources
+	}
+
+	type expectedResource struct {
+		name    string
+		ns      string
+		group   string
+		version string
+		kind    string
+	}
+	// Generate expected resources
+	var expectedResources = func(useCSR, enableOIDC bool) []expectedResource {
+		var resources = []expectedResource{
+			{name: render.PacketCaptureNamespace, ns: "", group: "", version: "v1", kind: "Namespace"},
+			{name: "pull-secret", ns: render.PacketCaptureNamespace, group: "", version: "v1", kind: "Secret"},
+			{name: render.PacketCaptureCertSecret, ns: render.PacketCaptureNamespace, group: "", version: "v1", kind: "Secret"},
+			{name: render.PacketCaptureServiceAccountName, ns: render.PacketCaptureNamespace, group: "", version: "v1", kind: "ServiceAccount"},
+			{name: render.PacketCaptureClusterRoleName, ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRole"},
+			{name: render.PacketCaptureClusterRoleBindingName, ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRoleBinding"},
+			{name: render.PacketCaptureDeploymentName, ns: render.PacketCaptureNamespace, group: "apps", version: "v1", kind: "Deployment"},
+			{name: render.PacketCaptureServiceName, ns: render.PacketCaptureNamespace, group: "", version: "v1", kind: "Service"},
+		}
+
+		if enableOIDC {
+			var oidc = []expectedResource{
+				{name: "tigera-dex-tls", ns: render.PacketCaptureNamespace, group: "", version: "v1", kind: "Secret"},
+			}
+			resources = append(resources, oidc...)
+		}
+
+		if useCSR {
+			resources = append(resources, expectedResource{"tigera-packetcapture:csr-creator", "", "rbac.authorization.k8s.io", "v1", "ClusterRoleBinding"})
+		}
+
+		return resources
+
+	}
+
+	// Generate expected environment variables
+	var expectedEnvVars = func(enableOIDC bool) []corev1.EnvVar {
+		var envVars = []corev1.EnvVar{
+			{Name: "PACKETCAPTURE_API_LOG_LEVEL", Value: "Info"},
+		}
+
+		if enableOIDC {
+			envVars = append(envVars, []corev1.EnvVar{
+				{
+					Name:  "PACKETCAPTURE_API_DEX_ENABLED",
+					Value: "true",
+				},
+				{
+					Name:  "PACKETCAPTURE_API_DEX_URL",
+					Value: "https://tigera-dex.tigera-dex.svc.cluster.local:5556/",
+				},
+				{
+					Name:  "PACKETCAPTURE_API_OIDC_AUTH_ENABLED",
+					Value: "true",
+				},
+				{
+					Name:  "PACKETCAPTURE_API_OIDC_AUTH_ISSUER",
+					Value: "https://127.0.0.1/dex",
+				},
+				{
+					Name:  "PACKETCAPTURE_API_OIDC_AUTH_JWKSURL",
+					Value: "https://tigera-dex.tigera-dex.svc.cluster.local:5556/dex/keys",
+				},
+				{
+					Name:  "PACKETCAPTURE_API_OIDC_AUTH_CLIENT_ID",
+					Value: "tigera-manager",
```

Comment:
@rene-dekker Is this value correct ? This seem what is being generated for a default dex installation.

Reply:
Yes, that is right

Reply:
See: https://github.com/tigera/operator/blob/fb4e8f655c0676b1f6f47076dcf7cefd0ff3650c/pkg/render/dex.go#L44

Reply:
Please use the convenience method. If something is missing, you can either add it or rectify in the other repo.
```
func (d *dexKeyValidatorConfig) RequiredEnv(prefix string) []corev1.EnvVar {
	return []corev1.EnvVar{
		{Name: fmt.Sprintf("%sDEX_ENABLED", prefix), Value: strconv.FormatBool(true)},
		{Name: fmt.Sprintf("%sDEX_ISSUER", prefix), Value: fmt.Sprintf("%s/dex", d.ManagerURI())},
		{Name: fmt.Sprintf("%sDEX_URL", prefix), Value: fmt.Sprintf("https://tigera-dex.tigera-dex.svc.%s:5556/", d.clusterDomain)},
		{Name: fmt.Sprintf("%sDEX_JWKS_URL", prefix), Value: fmt.Sprintf(jwksURI, d.clusterDomain)},
		{Name: fmt.Sprintf("%sDEX_CLIENT_ID", prefix), Value: DexClientId},
		{Name: fmt.Sprintf("%sDEX_USERNAME_CLAIM", prefix), Value: d.UsernameClaim()},
		{Name: fmt.Sprintf("%sDEX_GROUPS_CLAIM", prefix), Value: defaultGroupsClaim},
		{Name: fmt.Sprintf("%sDEX_USERNAME_PREFIX", prefix), Value: d.authentication.Spec.UsernamePrefix},
		{Name: fmt.Sprintf("%sDEX_GROUPS_PREFIX", prefix), Value: d.authentication.Spec.GroupsPrefix},
	}
}
```